server/src/main/java/org/elasticsearch/client/internal/AdminClient.java

```diff
@@ -14,15 +14,21 @@
  *
  * @see org.elasticsearch.client.internal.Client#admin()
  */
-public interface AdminClient {
+public class AdminClient {
 
-    /**
-     * A client allowing to perform actions/operations against the cluster.
-     */
-    ClusterAdminClient cluster();
+    protected final ClusterAdminClient clusterAdmin;
+    protected final IndicesAdminClient indicesAdmin;
```

Comment:
I think this could be a record. Do we rely on those fields being protected in lots of plases?

Reply:
There's mocks and other reflection-based stuff in tests that get in the way of tightening these things down. Possibly we could make them `private` but I'm hoping this stuff can all just go away soon so I don't think it's worth polishing this too much.